x/gamm/keeper/migrate.go

```diff
@@ -355,3 +357,57 @@ func (k Keeper) UpdateMigrationRecords(ctx sdk.Context, records []gammmigration.
 	}
 	return nil
 }
+
+// CreateClPoolAndLinkToCFMMPool creates a CL pool and links it to an existing CFMM pool.
+func (k Keeper) CreateClPoolAndLinkToCFMMPool(ctx sdk.Context, records []types.PoolRecordWithCFMMLink) error {
```

Comment:
this is a repetitive code from updrades in v16 `createCanonicalConcentratedLiquidityPoolAndMigrationLink`. Would love suggestion on how we can use the same function on upgrade folder as well as the keeper folder here

Reply:
The upgrade folder can just import functions in the keeper folder